developers/weaviate/configuration/roles.md

```diff
@@ -136,28 +136,435 @@ This example confers viewer permissions to a user for collections starting with
 
 </Tabs>
 
+import RolePyCode from '!!raw-loader!/_includes/code/python/howto.configure.rbac.roles.py';
+
 ## Role management
+To manage roles, first we must connect to the client with an API key that has access to do so, such as an admin API key.
+
+<Tabs groupId="languages">
+
+  <TabItem value="py" label="Python Client v4">
+    <FilteredTextBlock
+      text={RolePyCode}
+      startMarker="# START AdminClient"
+      endMarker="# END AdminClient"
+      language="py"
+    />
+  </TabItem>
+
+  <TabItem value="js" label="JS/TS Client v3">
+    TBD
+  </TabItem>
+
+  <TabItem value="go" label="Go">
+    TBD
+  </TabItem>
+
+  <TabItem value="java" label="Java">
+    TBD
+  </TabItem>
+
+</Tabs>
 
 ### Create a role
+To create a role, we provide the `role_name` and a list of permissions.
+For example, below we are creating a role called "devrel". 
+
+<Tabs groupId="languages">
+
+  <TabItem value="py" label="Python Client v4">
+    <FilteredTextBlock
+      text={RolePyCode}
+      startMarker="# START CreateRole"
+      endMarker="# END CreateRole"
+      language="py"
+    />
+  </TabItem>
+
+  <TabItem value="js" label="JS/TS Client v3">
+    TBD
+  </TabItem>
+
+  <TabItem value="go" label="Go">
+    TBD
+  </TabItem>
+
+  <TabItem value="java" label="Java">
+    TBD
+  </TabItem>
+
+</Tabs>
 
 ### Add permissions to a role
+There are two ways to add permissions to a role:
```

Comment:
I like the content, but this imo goes against the idea of a how-to page, as it forces people to read the examples top to bottom. 

I propose to have separate headers like:

> ### Create new role with permissions
> 
> This creates a "devrel" role with permissions to:
> - Read all collections that start with "Test_" 
> - Delete or create the collection "Test_DevRel"

And

> ### Grant additional permissions
> 
> This adds to the "devrel" role permissions to:
> - Read data in all collections that start with "Test_"
> - Create new data in "Test_DevRel"